src/connectors/h1_collect.py

```diff
@@ -202,73 +209,50 @@ def paginated_insert_reports(landing_table, options, dryrun):
     api_identifier = options['api_identifier']
     api_token = options['api_token']
     program_name = options['program_name']
+    next_exists = True
+    recorded_at = datetime.utcnow()
 
-    response = load_data(
-        'https://api.hackerone.com/v1/reports',
-        api_token,
-        api_identifier,
-        params={
-            'filter[program][]': program_name,
-            'page[size]': PAGE_SIZE,
-            'page[number]': page_number
-        },
-    )
-    reports = response.json()['data']
-    recorded_at = parsedate_to_datetime(response.headers.get('Date'))
-    # Insert 1st page
-    insert_reports(landing_table, reports, recorded_at, dryrun)
-
-    while 'next' in response.json()['links']:
-        page_number += 1
+    while next_exists:
         response = load_data(
             'https://api.hackerone.com/v1/reports',
             api_token,
             api_identifier,
             params={
-            'filter[program][]': program_name,
-            'page[size]': PAGE_SIZE,
-            'page[number]': page_number
+                'filter[program][]': program_name,
+                'page[size]': PAGE_SIZE,
+                'page[number]': page_number
             },
-        )   
+        )
+        next_exists = 'next' in response.json()['links']
+        page_number += 1
+
         reports = response.json()['data']
-        recorded_at = parsedate_to_datetime(response.headers.get('Date'))
-        # Insert new page
         insert_reports(landing_table, reports, recorded_at, dryrun)
 
-
+            
 def paginated_insert_transactions(landing_table, options, dryrun):
-    page_number = 1
     api_identifier = options['api_identifier']
     api_token = options['api_token']
     account_id = options['account_id']
 
-    response = load_data(
-        f'https://api.hackerone.com/v1/programs/{account_id}/billing/transactions',
-        api_token,
-        api_identifier,
-        params={
-            'page[size]': PAGE_SIZE,
-            'page[number]': page_number
-        },
-    )
-    transactions = response.json()['data']
-    recorded_at = parsedate_to_datetime(response.headers.get('Date'))
-    insert_transactions(landing_table, transactions, recorded_at, dryrun)
-
-    while 'next' in response.json()['links']:
```

Comment:
am I reading right that this loads a single page from each month? does this API not have pagination like the other one?

Reply:
correct - per https://api.hackerone.com/core-resources/#programs-get-payment-transactions "If you want to get transactions for an entire year, you will need to request each month individually."

Reply:
got it -- that might be worth linking for the explanation. it sounds like the resources are paginated, also, so it would likely make sense to move the pagination logic into the `load_data` function